### PR TITLE
Katello 3.11 Docs

### PR DESCRIPTION
**Pull request by jturel**
TODO:

- [x] Add release notes
- [x] Add API docs

### CODE REVIEW

**Comment by johnpmitsch** [comment]
From the installation page, is https://fedorapeople.org/groups/katello/releases/yum/3.11/katello/el7/x86_64/katello-repos-latest.rpm expected to work currently?

**Comment by ekohl** [comment]
> From the installation page, is https://fedorapeople.org/groups/katello/releases/yum/3.11/katello/el7/x86_64/katello-repos-latest.rpm expected to work currently?

No, only after the release has gone through the pipeline and verified to work it's pushed to the public location.

**Comment by johnpmitsch** [comment]
The URL doesn't look correct for the client installation
`yum install -y https://yum.theforeman.org/client//el7/x86_64/foreman-client-release.rpm`, will that also be fixed when things are out of the RC phase?

**Comment by johnpmitsch** [comment]
Other than that open question, things look good! :+1:

**Comment by jturel** [comment]
@johnpmitsch nice catch. The page was missing foreman_version. If you leave the container running and just pull down the changes it'll regenerate the HTML (saves time from bringing up the container again)

**Comment by johnpmitsch** [comment]
The client links look much better now, but I am not able to access the sles11 and 12 links? i.e. https://yum.theforeman.org/client/1.21/sles11/x86_64/foreman-client-release.rpm, it looks like that rpm is not in that folder. I'm not sure if that is something that happens later, but figured I would ask since it is there for other distros

**Comment by jturel** [comment]
I see what you mean. Looks like the same problem is present for 1.20 as well - for that reason I'd like to treat that as a separate problem especially since I don't know how to fix it at the moment. Will make a note

**Review by johnpmitsch — APPROVED** [review]
Things look good aside from the SLES links, which sound like they are being addressed separately

**Comment by ekohl** [comment]
The problem is that our CI pipeline doesn't publish sles: https://ci.theforeman.org/blue/organizations/jenkins/foreman-client-1.21-rpm-pipeline/detail/foreman-client-1.21-rpm-pipeline/3/pipeline (same for 1.20 I think because it was copied).

File: _includes/plugins/katello/sidebar_3.11.html [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+    <li class="item">
+      <a href="/plugins/katello/3.11/upgrade/index.html">2.1 Katello</a>
+    </li>
+    <li class="item">
+      <a href="/plugins/katello/3.11/upgrade/smart_proxy.html">2.2 Smart proxy</a>
+    </li>
+    <li class="item">
+      <a href="/plugins/katello/3.11/upgrade/clients.html">2.3 Clients</a>
+    </li>
+    {% if page.version == 'nightly' or page.version == 3.2 %}
```
Comment (evgeni) [inline _includes/plugins/katello/sidebar_3.11.html]: I think we can drop the puppet upgrade that was needed in 3.2 now? :)

Comment (evgeni) [reply _includes/plugins/katello/sidebar_3.11.html]: (doesn't have to be in this PR, we'd be updating other versioned pages too then anyways)

File: _includes/plugins/katello/sidebar_3.11.html [hunk 2]
```diff
@@ -0,0 +1,180 @@
+<ul class="header nav nav-list bs-docs-sidenav">
+  <li class="item">
+    <a>1. Installation</a>
+  </li>
+
+  <ul class="sub-header-static nav nav-list bs-docs-sidenav">
+    <li class="item">
+      <a href="/plugins/katello/3.11/installation/index.html">1.1 Katello</a>
```
Comment (ekohl) [inline _includes/plugins/katello/sidebar_3.11.html]: Every use of `3.11`: can it be `{{ page.version }}` as well?

Comment (jturel) [reply _includes/plugins/katello/sidebar_3.11.html]: fixed for 3.11 and nightly

File: _includes/plugins/katello/sidebar_3.11.html (hunk 1 above)
Comment (jturel) [reply _includes/plugins/katello/sidebar_3.11.html]: removed this check for 3.11 and nightly & removed upgrades/puppet.md for > 3.2

**Review by ehelms — APPROVED** [review]

**Comment by jturel** [comment]
@ehelms care to merge? :)

**Pull request merged**